src/test/scala/strawman/collection/test/Test.scala

```diff
@@ -469,6 +469,8 @@ class StrawmanTest {
     val xs9t: mutable.HashMap[Int, String] = xs9
     val xs10 = xs1 ++ xs2
     val xs11: immutable.SortedMap[String, Int] = xs10
+    val xs12 = xs11.collect({ case (k, v) if k.length == v => (v, k) }: scala.PartialFunction[(String, Int), (Int, String)])
```

Comment:
Simplified:

```scala
trait SortedMap[K, V] {
  def collect[B](pf: PartialFunction[(K, V), B]): Int = 0
  def collect[K2 : Ordering, V2](pf: PartialFunction[(K, V), (K2, V2)]): Int = 0

  def foo(xs: SortedMap[String, scala.Int]) = xs.collect { case (k, v) => 1 }
}
```
It works with `Function1` instead of `PartialFunction`. Could be related to 9cd3becde5bb7c37af64ad5ac1bd15aa4fafd1b4.